### ISSUE TITLE
problem with unsatisfactory sampling result

### ISSUE THREAD

**Issue by MaybeRichard** [mazurowski-lab/segmentation-guided-diffusion#16]
I've read some issues and I met the same problem with unsatisfactory sampling result, such as a lot of noisy in the result. Currently, I'm using DDIM as scheduler and I'm trying to change to DDPM. The dataset i'm using only have 110 paired images and masks and I also tried other models like SPADE, Retree, LDM etc and they all have the same problem. I'm not sure if it is the problem with the size of dataset, so can I discuss with you about this problem?

I also tried other solution to improve the sampling result:

- enlarge the resoltuion from 256 to 512, and the result gets better but still with noise.

- pretrain with large dataset without mask label and using Gaussian noise as mask label(to learn the distribution of specific types of image) and then use small size of paired image-mask data finetune.

that's all I've tried and I think this is meaningful because there is a lot of domain lack of enough labeled dataset.

Looking forward to your reply and thanks for your contribution for medical imageing filed!

<div style="display: flex; justify-content: space-around;">
    <figure>
        <img src="https://github.com/user-attachments/assets/c85aa754-37c7-4b2e-8213-fc3c9bbfb79c" width="400" alt="Sampling Image 1">
        <figcaption>Figure 1: Sampling Image</figcaption>
    </figure>
    <figure>
        <img src="https://github.com/user-attachments/assets/d6bd986f-141d-47b0-bb72-5f5efcc6b94d" width="400" alt="Real Image 1">
        <figcaption>Figure 2: Real Image</figcaption>
    </figure>
</div>

**Comment by nickk124** [comment]
Hello,

Unfortunately, your issue likely is the small dataset size, especially because you're facing the same problems with other models. (I'm assuming you mean 110 2D images, not many more 2D images taken from 110 3D volumes.) In our experiments, the training sets had at least a few thousand images + masks.

It looks like you're working with OCT images. As you alluded to, one option which may work for you could be to:
1.  pretrain an LDM autoencoder on a bunch of unlabeled OCT images from some other dataset (I know there are larger public OCT datasets out there)
2. With the autoencoder trained, train the LDM itself conditioned on masks with your smaller training set.

The assumption here is that with a good latent representation learned from the larger dataset, it hopefully may require fewer examples to learn the mask conditioning.

I hope that was helpful!

**Comment by MaybeRichard** [comment]
Really appreciate your comment. I do have only 110 2D OCT images and I'ii try to pretrain current VAE with larger datasets. Thanks again!

**Comment by nickk124** [comment]
No problem!
